ingredient/ingredient-app-insights/src/functions.ts

```diff
@@ -1,12 +1,35 @@
-import {BaseUtility, IngredientManager} from '@azbake/core'
+import { BaseUtility, IngredientManager } from '@azbake/core'
+import { ApplicationInsightsManagementClient } from '@azure/arm-appinsights'
+import { utils } from 'mocha';
 
 export class AppInsightsUtils extends BaseUtility {
-    
-    public create_resource_name(): string {
+
+    public get_resource_name(shortName: string): string {
         let util = IngredientManager.getIngredientFunction("coreutils", this.context)
 
-        const name = util.create_resource_name("ai", null, true);
-        return name;
+        //App Insights telemetry will be centralized.  Therefore don't appenda  region code.
+        const appInsightsName = util.create_resource_name("ai", shortName, false);
+        return appInsightsName;
     }
-}
 
+    public async get_instrumentation_key(shortName: string, rgShortName: string | null = null): Promise<string> {
+
+        let util = IngredientManager.getIngredientFunction("coreutils", this.context)
+        
+        //App Insights telemetry will be centralized.  Therefore don't append a region code.  
+        //Also, pass in shortname rather than using default.  The default is the package name being deployed but we want the App Insights resource name/group.
+        const aiName = util.create_resource_name("ai", shortName, false);
+        
+        const rgName: string = await util.resource_group(shortName, false);
```

Comment:
should this be rgShortName?

Reply:
good eyes.  not sure what happened...  I'll fix.